packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -25,17 +25,31 @@ export class DataSourceWithBackend<
    * Ideally final -- any other implementation would be wrong!
    */
   query(request: DataQueryRequest): Observable<DataQueryResponse> {
-    const { targets, intervalMs, maxDataPoints, range } = request;
+    const { targets, intervalMs, maxDataPoints, range, requestId } = request;
+    let rangeFrom: any = undefined;
+    let rangeTo: any = undefined;
+
+    if (range) {
+      rangeFrom = range.from.valueOf().toString();
+      rangeTo = range.to.valueOf().toString();
```

Comment:
are these supposed to be numbers as strings?  I would rather we just end numbers no?

Reply:
Haven't thought about changing that since we always handled it this way, but if you want to change that I'm open ears. Could have been that it historically allowed both unix and string value like "now-1h"? We use timerange parser in the backend https://github.com/grafana/grafana/blob/master/pkg/api/metrics.go#L100 so no biggie from my perspective.

Reply:
While we are chaning things... why not just have the request match the final proto structure:
https://github.com/grafana/grafana-plugin-sdk-go/blob/master/proto/backend.proto#L74